PublicMapping/districtbuilder
# Edit project name

## Overview

Adds ability to edit project names inline in the project screen header

### Checklist


### Demo

when the page is loaded:
![localhost_3003_projects_e1cb7dfc-eafb-4566-bbec-9c325aae58ce (4)](https://user-images.githubusercontent.com/4432106/97007373-4f0ee100-150f-11eb-937a-93dc6dde5591.png)

editing:
![localhost_3003_projects_e1cb7dfc-eafb-4566-bbec-9c325aae58ce (3)](https://user-images.githubusercontent.com/4432106/97007374-4f0ee100-150f-11eb-9925-397381f769ab.png)

save button disabled when name is empty:
![localhost_3003_projects_e1cb7dfc-eafb-4566-bbec-9c325aae58ce (5)](https://user-images.githubusercontent.com/4432106/97007370-4e764a80-150f-11eb-9058-ddec6a3dc5c9.png)

### Notes

This is based on top of #484, I'll retarget it to `develop` once that is merged.

## Testing Instructions

- `scripts/server`
- Ensure you can edit project names
- Ensure that reloading the project screen shows your changes have persisted
- Attempt to set a project name to an empty string - the save button should be disabled
- The project name should continue to be read-only and show the map creator's name when viewing another user's project

Closes #220
